website/addons/owncloud/views.py

```diff
@@ -0,0 +1,173 @@
+"""Views for the node settings page."""
+# -*- coding: utf-8 -*-
+import httplib as http
+
+from flask import request
+from modularodm import Q
+from modularodm.storage.base import KeyExistsException
+
+from framework.auth.decorators import must_be_logged_in
+
+from website.addons.base import generic_views
+from owncloud import Client as OwnCloudClient
+from website.addons.owncloud.model import OwnCloudProvider
+from website.addons.owncloud.settings import DEFAULT_HOSTS
+from website.addons.owncloud.serializer import OwnCloudSerializer
+from website.oauth.models import ExternalAccount
+from website.project.decorators import (
+    must_have_addon)
+from website.util import rubeus, api_url_for
+
+from website.addons.owncloud.utils import ExternalAccountConverter
+
+SHORT_NAME = 'owncloud'
+FULL_NAME = 'OwnCloud'
+
+owncloud_account_list = generic_views.account_list(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+owncloud_import_auth = generic_views.import_auth(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+owncloud_deauthorize_node = generic_views.deauthorize_node(
+    SHORT_NAME
+)
+
+owncloud_get_config = generic_views.get_config(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+def owncloud_root_folder(node_settings, auth, **kwargs):
+    """Build HGrid JSON for root node. Note: include node URLs for client-side
+    URL creation for uploaded files.
+    """
+    node = node_settings.owner
+    root = rubeus.build_addon_root(
+        node_settings=node_settings,
+        name=node_settings.folder_name,
+        permissions=auth,
+        user=auth.user,
+        nodeUrl=node.url,
+        nodeApiUrl=node.api_url,
+    )
+    return [root]
+
+@must_be_logged_in
+def owncloud_user_config_get(auth, **kwargs):
+    """View for getting a JSON representation of the logged-in user's
+    OwnCloud user settings.
+    """
+
+    user_addon = auth.user.get_addon('owncloud')
+    user_has_auth = False
+    if user_addon:
+        user_has_auth = user_addon.has_auth
+
+    return {
+        'result': {
+            'userHasAuth': user_has_auth,
+            'urls': {
+                'create': api_url_for('owncloud_add_user_account'),
+                'accounts': api_url_for('owncloud_account_list'),
+            },
+            'hosts': DEFAULT_HOSTS,
+        },
+    }, http.OK
+
+
+## Config ##
+
+@must_be_logged_in
+def owncloud_add_user_account(auth, **kwargs):
+    """Verifies new external account credentials and adds to user's list"""
+    user = auth.user
+    provider = OwnCloudProvider()
+
+    host = request.json.get('host').rstrip('/')
+    username = request.json.get('username')
+    password = request.json.get('password')
+
+    oc = OwnCloudClient(host, verify_certs=False)
```

Comment:
@kwierman, do you have any guess as to how common or uncommon it would be for any given ownCloud user to not enable SSL on a server they've put up? For something that only accepts Basic Auth, it seems like using SSL would be the norm rather than the exception, but I might have some biases at play here. Regardless, we _cannot_ allow plaintext passwords to be transmitted over the network in a production environment -- it's just too much of a security risk.

It seems like the right UX here would be to:
* have SSL required
* an error raised here if it's not enabled for a `host`, causing users to be presented a message about encryption being a good thing, and possibly directing them to [ownCloud documentation about secure encryption practices](https://doc.owncloud.org/server/9.1/admin_manual/configuration_server/harden_server.html?highlight=ssl#use-https)

Pinging @saradbowman or @nicipfeiffer about UX / language